notebooks/NIRISS_WFSS_advanced/02_niriss_wfss_spec2.ipynb

```diff
@@ -0,0 +1,1146 @@
+    "source_id = 118\n",
+    "wh_x1d_118, wh_cal_118 = find_source_ext(x1d_hdu, cal_hdu, source_id)\n",
+    "\n",
+    "x1d_data_118 = x1d_hdu[wh_x1d_118].data \n",
+    "cal_data_118 = cal_hdu[wh_cal_118].data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a362fa9f-ae33-4dcc-954c-471a80b41d37",
+   "metadata": {},
+   "source": [
+    "First, let's look at the extraction box as it appears on the rate image. This will help us get a global feel for how much we can trust the pipeline's extraction. \n",
+    "\n",
+    "*Note: In this example, the extraction box isn't fully cenetered around the spectrum. There is an ongoing calibration effort to better account for difference in the spectral trace shape across the NIRISS detector. Updates about the status of this calibration can be found on the [NIRISS jdox](https://jwst-docs.stsci.edu/jwst-calibration-pipeline-caveats/jwst-wide-field-slitless-spectroscopy-pipeline-caveats#JWSTWideFieldSlitlessSpectroscopyPipelineCaveats-NIRISSWFSS)*"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ddc84d02-8b67-440b-9380-53d5a27a5349",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# fill in the nan values from the bad pixels with zero to make it easier to look at\n",
+    "rate_data = rate_hdu[1].data\n",
+    "rate_data[np.isnan(rate_data)] = 0\n",
+    "\n",
+    "# extraction box parameters from the header of the cal data:\n",
+    "cal_header = cal_hdu[wh_cal_118].header\n",
+    "sx_left = cal_header['SLTSTRT1']\n",
+    "swidth = cal_header['SLTSIZE1']\n",
+    "sx_right = cal_header['SLTSTRT1'] + swidth\n",
+    "sy_bottom = cal_header['SLTSTRT2']\n",
+    "sheight = cal_header['SLTSIZE2']\n",
+    "sy_top = cal_header['SLTSTRT2'] + sheight\n",
+    "\n",
+    "# plot the rate file and the extraction box\n",
+    "fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(13, 5))\n",
+    "ax1.imshow(rate_data, origin='lower', vmin=0, vmax=2, aspect='auto')\n",
+    "ax2.imshow(rate_data, origin='lower', vmin=0, vmax=2, aspect='auto')\n",
+    "\n",
+    "rectangle = patches.Rectangle((sx_left, sy_bottom), swidth, sheight, edgecolor='darkorange', facecolor=\"None\", linewidth=1)\n",
+    "ax1.add_patch(rectangle)\n",
+    "ax1.set_title(rate_file)\n",
+    "\n",
+    "rectangle2 = patches.Rectangle((sx_left, sy_bottom), swidth, sheight, edgecolor='darkorange', facecolor=\"None\", linewidth=2)\n",
+    "ax2.add_patch(rectangle2)\n",
+    "ax2.set_xlim(sx_left-30, sx_right+30)\n",
+    "ax2.set_ylim(sy_bottom-30, sy_top+30)\n",
+    "ax2.set_title(f'Source {source_id} Zoom in')\n",
+    "\n",
+    "plt.suptitle(f\"{cal_hdu[0].header['FILTER']} {cal_hdu[0].header['PUPIL']}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "22a7c98b-0a1b-4d1f-9323-25eb08696e4d",
+   "metadata": {},
+   "source": [
+    "We can then take a look at the extracted spectrum in this box both in the cal file and the x1d file. **Information about this source!** \n",
```

Comment:
@camipacifici would you be able to provide details about the source you picked? Steph had this suggestion, and I'm not actually 100% sure what source I'm looking at or what lines there are.